libfwupd/fwupd-client.c

```diff
@@ -4053,16 +4053,18 @@ fwupd_client_refresh_remote_signature_cb(GObject *source, GAsyncResult *res, gpo
 	}
 
 	/* is the signature checksum the same? */
-	checksum_kind = fwupd_checksum_guess_kind(fwupd_remote_get_checksum(data->remote));
-	checksum =
-	    g_compute_checksum_for_data(checksum_kind,
-					(const guchar *)g_bytes_get_data(data->signature, NULL),
-					g_bytes_get_size(data->signature));
-	if (g_strcmp0(checksum, fwupd_remote_get_checksum(data->remote)) == 0) {
-		g_info("metadata signature of %s is unchanged, skipping",
-		       fwupd_remote_get_id(data->remote));
-		g_task_return_boolean(task, TRUE);
-		return;
+	if (fwupd_remote_get_checksum(data->remote) != NULL) {
+		checksum_kind = fwupd_checksum_guess_kind(fwupd_remote_get_checksum(data->remote));
```

Comment:
Any variable defs we can move into the inner block?

Reply:
moved